asset/mock.go

```diff
@@ -205,6 +206,98 @@ func (m *MockGroupTxBuilder) BuildGenesisTx(newAsset *Asset) (*wire.MsgTx,
 // GenesisTxBuilder interface.
 var _ GenesisTxBuilder = (*MockGroupTxBuilder)(nil)
 
+type storedTapTree struct {
+	branches bool
+	treeData [][]byte
+}
+
+type MockTapscriptTreeStore struct {
+	Store               map[chainhash.Hash]storedTapTree
+	FailLoad, FailStore bool
+}
+
+func NewMockTapscriptTreeStore() *MockTapscriptTreeStore {
+	return &MockTapscriptTreeStore{
+		Store: make(map[chainhash.Hash]storedTapTree),
+	}
+}
+
+func (m *MockTapscriptTreeStore) StoreTapscriptTree(_ context.Context,
+	tapTree TapscriptTreeNodes) (*chainhash.Hash, error) {
+
+	if m.FailStore {
+		return nil, fmt.Errorf("failed to store tapscript tree")
+	}
+
+	var rootHash chainhash.Hash
+
+	tapLeaves := tapTree.GetLeaves().UnwrapToPtr()
```

Comment:
Wouldn't this be a prime example for `WhenSome()`?
Although I have to say I haven't fully warmed up to that functional style of `Option` and `Either`, as it breaks one of the Golang patterns I've become very fond of: The happy path is non-indented in most cases...

But probably not something to discuss on this PR specifically.

Reply:
The problem I had with `WhenSome` was that it requires any logic using the value inside the `Option`, to be rewritten as a standalone function with side effects. Which is doable but reads awkwardly imo, for the changes needed in this PR at least.